Opsctrlmonitoring development

ianrscottexp opened the pull request:
Updated code to remove code that wasn't needed and add agent list yml to satisfy request

exp-hc commented:
You can keep the same pull request open and make additional commits to the existing branch moving forward. I'll take a look at the changes this afternoon.

exp-hc commented on plugins/modules/elastic_pkgpolicy_info.py:
```diff
@@ -0,0 +1,100 @@
+def main():
+
+    module_args=dict(   
+        host=dict(type='str',Required=True),
+        port=dict(type='int', default=9243),
+        username=dict(type='str', Required=True),
+        password=dict(type='str', no_log=True, Required=True),   
+        verify_ssl_cert=dict(type='bool', default=True),
+        agent_policy_name=dict(type='str'),
+        agent_policy_id=dict(type='str'),
+        integration_name=dict(type='str', Required=True),
+        pkg_policy_name=dict(type='str', Required=True)
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_pkgpolicy_info.py:
```diff
@@ -0,0 +1,100 @@
+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),
+        #('alert-type', 'metrics_threshold', ('conditions'))
+    
+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)
```
Since a name or id is required as an input, the following logic would ensure the requirement is met. 

```
module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,
                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],
                           required_one_of=[('agent_policy_name', 'agent_policy_id')])
```

exp-hc commented on plugins/module_utils/kibana.py:
```diff
@@ -70,4 +71,210 @@ def get_alert_connector_by_name(self, connector_name):
+    return(pkg_policy_object)
+  
+  def create_pkg_policy(self,agent_policy_id, integration_object):
+    self.pkg_policy_name = self.module.params.get('pkg_policy_name')
+    self.pkg_policy_desc = self.module.params.get('pkg_policy_desc')
```
Module parameters should be gathered in the module or in `init` if in module_utils as mentioned above.

exp-hc commented on plugins/module_utils/kibana.py:
```diff
@@ -70,4 +71,210 @@ def get_alert_connector_by_name(self, connector_name):
+
+  def create_agent_policy(self, check_mode=False):
+    agent_policy_action = Kibana(self.module)
+    self.agent_policy_id = self.module.params.get('agent_policy_id')
+    self.agent_policy_name = self.module.params.get('agent_policy_name')
+    self.agent_policy_desc = self.module.params.get('agent_policy_desc')
```
Module parameters should be gathered in the module or in `init` if in module_utils as mentioned above.

exp-hc commented on plugins/module_utils/kibana.py:
```diff
@@ -70,4 +71,210 @@ def get_alert_connector_by_name(self, connector_name):
+  def get_agent_policy_byname(self):
+    agent_policy_action = Kibana(self.module)
+    agent_policy_object = ""
+    self.agent_policy_name=self.module.params.get('agent_policy_name')
```
Module parameters should be gathered in the module or in `init` if in module_utils as mentioned above.

exp-hc commented on plugins/module_utils/kibana.py:
```diff
@@ -70,4 +71,210 @@ def get_alert_connector_by_name(self, connector_name):
+      return
+  
+  def get_agent_policy_byid(self):
+    self.agent_policy_id = self.module.params.get('agent_policy_id')
```
Module parameters should be gathered in the module or in `init` if in module_utils as mentioned above.

exp-hc commented on plugins/modules/elastic_agentlist_info.py:
```diff
@@ -0,0 +1,56 @@
+def main():
+
+    module_args=dict(    
+        host=dict(type='str',Required=True),
+        port=dict(type='int', default=9243),
+        username=dict(type='str', Required=True),
+        password=dict(type='str', no_log=True, Required=True),   
+        verify_ssl_cert=dict(type='bool', default=True)
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_agentpolicy.py:
```diff
@@ -0,0 +1,70 @@
+def main():
+
+    module_args=dict(    
+        host=dict(type='str',Required=True),
+        port=dict(type='int', default=9243),
+        username=dict(type='str', Required=True),
+        password=dict(type='str', no_log=True, Required=True),   
+        verify_ssl_cert=dict(type='bool', default=True),
+        agent_policy_name=dict(type='str', Required=True),
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_agentpolicy_info.py:
```diff
@@ -0,0 +1,61 @@
+def main():
+
+    module_args=dict(    
+        host=dict(type='str',Required=True),
+        port=dict(type='int', default=9243),
+        username=dict(type='str', Required=True),
+        password=dict(type='str', no_log=True, Required=True),   
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -0,0 +1,172 @@
+def main():
+
+    module_args=dict(   
+        host=dict(type='str',Required=True),
+        port=dict(type='int', default=9243),
+        username=dict(type='str', Required=True),
+        password=dict(type='str', no_log=True, Required=True),   
+        verify_ssl_cert=dict(type='bool', default=True),
+        agent_policy_id=dict(type='str'),
+        agent_policy_name=dict(type='str'),
+        integration_name=dict(type='str', Required=True),
+        pkg_policy_name=dict(type='str', Required=True),
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_pkgpolicy.py:
```diff
@@ -0,0 +1,109 @@
+def main():
+
+    module_args=dict(   
+        host=dict(type='str',Required=True),
+        port=dict(type='int', default=9243),
+        username=dict(type='str', Required=True),
+        password=dict(type='str', no_log=True, Required=True),   
+        verify_ssl_cert=dict(type='bool', default=True),
+        agent_policy_name=dict(type='str'),
+        agent_policy_id=dict(type='str'),
+        integration_name=dict(type='str', Required=True),
+        pkg_policy_name=dict(type='str', Required=True),
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_pkgpolicy_info.py:
```diff
@@ -0,0 +1,100 @@
+from ansible.module_utils.basic import _ANSIBLE_ARGS, AnsibleModule
+#from ansible.module_utils.basic import *
+
+import json
+  
+try:
+  from ansible_collections.expedient.elastic.plugins.module_utils.ece import ECE
+except:
+  import sys
+  import os
+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'
+  sys.path.append(util_path)
+  from ece import ECE
+
```
ECE is imported but never used

exp-hc commented on plugins/modules/elastic_pkgpolicy_info.py:
```diff
@@ -0,0 +1,100 @@
+from ansible.module_utils.basic import _ANSIBLE_ARGS, AnsibleModule
+#from ansible.module_utils.basic import *
+
+import json
```
json is imported but never used

exp-hc commented on plugins/modules/elastic_pkgpolicy.py:
```diff
@@ -0,0 +1,109 @@
+from ansible.module_utils.basic import _ANSIBLE_ARGS, AnsibleModule
+#from ansible.module_utils.basic import *
+  
+try:
+  from ansible_collections.expedient.elastic.plugins.module_utils.ece import ECE
+except:
+  import sys
+  import os
+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'
+  sys.path.append(util_path)
+  from ece import ECE
```
ECE is imported but never used

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -0,0 +1,172 @@
+    else:
+        results['changed'] = True
+        
+    kibana = Kibana(module)
```
In order to get the params defined in `SecurityBaseline` `init`, `Kibana` should be replaced here with `SecurityBaseline`

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -0,0 +1,172 @@
+                pkg_policy_object.pop('updated_by')
+                break
+            i=+1
+          pkg_policy_object_json = json.dumps(pkg_policy_object)
+          pkg_policy_update = kibana.update_pkg_policy(pkg_policy_object_id,pkg_policy_object_json)
```
The `send_api_request` takes care of `json.dumps` so this isn't needed.

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -0,0 +1,172 @@
+from ansible.module_utils.basic import _ANSIBLE_ARGS, AnsibleModule
+import json
+
+try:
+  from ansible_collections.expedient.elastic.plugins.module_utils.ece import ECE
+except:
+  import sys
+  import os
+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'
+  sys.path.append(util_path)
+  from ece import ECE
```
ECE is imported but never used

exp-hc commented on plugins/modules/elastic_pkgpolicy.py:
```diff
@@ -0,0 +1,109 @@
+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),
+        #('alert-type', 'metrics_threshold', ('conditions'))
+    
+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)
```
Since a name or id is required as an input, the following logic would ensure the requirement is met. 

```
module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,
                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],
                           required_one_of=[('agent_policy_name', 'agent_policy_id')])
```

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -0,0 +1,172 @@
+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),
+        #('alert-type', 'metrics_threshold', ('conditions'))
+    
+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)
```
Since a name or id is required as an input, the following logic would ensure the requirement is met. 

```
module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,
                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],
                           required_one_of=[('agent_policy_name', 'agent_policy_id')])
```

exp-hc commented on plugins/modules/elastic_agentpolicy_info.py:
```diff
@@ -0,0 +1,61 @@
+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),
+        #('alert-type', 'metrics_threshold', ('conditions'))
+    
+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)
```
Since a name or id is required as an input, the following logic would ensure the requirement is met. 

```
module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,
                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],
                           required_one_of=[('agent_policy_name', 'agent_policy_id')])
```

exp-hc reviewed (changes requested):
I've made several comments for items I've found that look off.

ianrscottexp commented:
I believe I have addressed all your comments and updated the branch.

From: Hunter Christain ***@***.***>
Sent: Wednesday, December 8, 2021 12:31 PM
To: Expedient/ansible-collection-elastic ***@***.***>
Cc: Scott, Ian ***@***.***>; Author ***@***.***>
Subject: Re: [Expedient/ansible-collection-elastic] Opsctrlmonitoring development (PR #8)

[ CAUTION: External Email ]


@exp-hc requested changes on this pull request.

I've made several comments for items I've found that look off.

________________________________

In plugins/module_utils/kibana.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765090071>:

> +  def update_rule(self, body):

+    endpoint = "detection_engine/rules"

+    update_rule = self.send_api_request(endpoint,'PATCH',data=body)

+    return update_rule

+

+  def get_rules(self,page_size,page_no):

+    endpoint = "detection_engine/rules/_find?page=" + str(page_no) + "&per_page=" + str(page_size)

+    rules = self.send_api_request(endpoint, 'GET')

+    return rules

+

+  def enable_rule(self,rule_id):

+    body={

+      'enabled': True,

+      'id': rule_id

+    }

+    update_rule = Kibana.update_rule(self, body)

Any reference to Kibana within the Kibana class needs to be changed to self. This occurs several time in these updates but I'm flagging the first instance.

________________________________

In plugins/module_utils/kibana.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765091605>:

> +

+  def get_all_pkg_policies(self):

+      endpoint  = 'fleet/package_policies'

+      pkgpolicy_objects = self.send_api_request(endpoint, 'GET')

+      return pkgpolicy_objects

+

+  def update_pkg_policy(self,pkgpolicy_id,body):

+      endpoint = "fleet/package_policies/" + pkgpolicy_id

+      if not self.module.check_mode:

+        pkg_policy_update = self.send_api_request(endpoint, 'PUT', data=body)

+      else:

+        pkg_policy_update = "Cannot proceed with check_mode set to " + self.module.check_mode

+      return pkg_policy_update

+

+  def get_pkg_policy(self,agent_policy_id):

+    self.integration_name = self.module.params.get('integration_name')

Module parameters should be gathered in the module or in init if in module_utils. For this example, this should be moved to the modules calling this functions and passed to get_pkg_policy as a variable.

________________________________

In plugins/module_utils/kibana.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765094550>:

> +      body_JSON = dumps(body)

+      endpoint = 'fleet/package_policies'

+      if not self.module.check_mode:

+        pkg_policy_object = self.send_api_request(endpoint, 'POST', data=body_JSON)

+      else:

+        pkg_policy_object = "Cannot proceed with check_mode set to " + self.module.check_mode

+      return pkg_policy_object

+    else:

+      return pkg_policy_object

+

+# Elastic Agent Policy functions

+

+  def get_all_agent_policys(self):

+    endpoint  = 'fleet/agent_policies'

+    agent_policy_objects = self.send_api_request(endpoint, 'GET')

+    return(agent_policy_objects)

I've never seen return used this way in python, it looks this PR has it in multiple formats, we should stick with return <return data> or in this example return agent_policy_objects as a standard

________________________________

In plugins/modules/ece_cluster_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765096688>:

> +try:

+  from ansible_collections.expedient.elastic.plugins.module_utils.elastic import Elastic

+except:

+  import sys

+  import os

+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'

+  sys.path.append(util_path)

+  from elastic import Elastic

+

+try:

+  from ansible_collections.expedient.elastic.plugins.module_utils.kibana import Kibana

+except:

+  import sys

+  import os

+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'

+  sys.path.append(util_path)

+  from kibana import Kibana

Elastic and Kibana are not used in this module so they can be removed from the import.

________________________________

In plugins/modules/ece_cluster_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765098330>:

> +        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        deployment_name=dict(type='str', Required=True)

Required should be required

________________________________

In plugins/modules/ece_cluster_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765098606>:

> +##################################################################

+#

+#   Script to get Elastic Deployment ID and object from Elastic Admin

+#

+#   Version 1.0 - 11/17/2021 - Ian Scott - Initial Draft

+#

+##################################################################

This needs updated to match the other modules.

________________________________

In plugins/modules/elastic_pkgpolicy_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765098936>:

> +        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_name=dict(type='str'),

+        agent_policy_id=dict(type='str'),

+        integration_name=dict(type='str', Required=True),

+        pkg_policy_name=dict(type='str', Required=True)

Required should be required

________________________________

In plugins/modules/elastic_pkgpolicy_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765100573>:

> +    module_args=dict(

+        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_name=dict(type='str'),

+        agent_policy_id=dict(type='str'),

+        integration_name=dict(type='str', Required=True),

+        pkg_policy_name=dict(type='str', Required=True)

+    )

+    argument_dependencies = []

+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),

+        #('alert-type', 'metrics_threshold', ('conditions'))

+

+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)

Since a name or id is required as an input, the following logic would ensure the requirement is met.

module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,

                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],

                           required_one_of=[('agent_policy_name', 'agent_policy_id')])

________________________________

In plugins/module_utils/kibana.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765101941>:

> +    self.pkg_policy_name = self.module.params.get('pkg_policy_name')

+    self.pkg_policy_desc = self.module.params.get('pkg_policy_desc')

Module parameters should be gathered in the module or in init if in module_utils as mentioned above.

________________________________

In plugins/module_utils/kibana.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765102011>:

> +    self.agent_policy_id = self.module.params.get('agent_policy_id')

+    self.agent_policy_name = self.module.params.get('agent_policy_name')

+    self.agent_policy_desc = self.module.params.get('agent_policy_desc')

Module parameters should be gathered in the module or in init if in module_utils as mentioned above.

________________________________

In plugins/module_utils/kibana.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765102065>:

> +      body_JSON = dumps(body)

+

+      if not self.module.check_mode:

+        endpoint  = 'fleet/agent_policies'

+        agent_policy_object = self.send_api_request(endpoint, 'POST', data=body_JSON)

+        agent_policy_object = agent_policy_object['item']

+      else:

+        agent_policy_object = "Cannot proceed with check_mode set to " + self.module.check_mode

+    else:

+      return

+    return(agent_policy_object)

+

+  def get_agent_policy_byname(self):

+    agent_policy_action = Kibana(self.module)

+    agent_policy_object = ""

+    self.agent_policy_name=self.module.params.get('agent_policy_name')

Module parameters should be gathered in the module or in init if in module_utils as mentioned above.

________________________________

In plugins/module_utils/kibana.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765102155>:

> +  def get_agent_policy_byname(self):

+    agent_policy_action = Kibana(self.module)

+    agent_policy_object = ""

+    self.agent_policy_name=self.module.params.get('agent_policy_name')

+    agent_policy_objects = agent_policy_action.get_all_agent_policys()

+    for agent_policy in agent_policy_objects['items']:

+        if agent_policy['name'] == self.agent_policy_name:

+            agent_policy_object = agent_policy

+            continue

+    if agent_policy_object:

+      return(agent_policy_object)

+    else:

+      return

+

+  def get_agent_policy_byid(self):

+    self.agent_policy_id = self.module.params.get('agent_policy_id')

Module parameters should be gathered in the module or in init if in module_utils as mentioned above.

________________________________

In plugins/modules/elastic_agentlist_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765102810>:

> +        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True)

Required should be required

________________________________

In plugins/modules/elastic_agentpolicy.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765102883>:

> +        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_name=dict(type='str', Required=True),

Required should be required

________________________________

In plugins/modules/elastic_agentpolicy_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765102940>:

> +        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

Required should be required

________________________________

In plugins/modules/elastic_endpoint_security.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765103185>:

> +        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_id=dict(type='str'),

+        agent_policy_name=dict(type='str'),

+        integration_name=dict(type='str', Required=True),

+        pkg_policy_name=dict(type='str', Required=True),

Required should be required

________________________________

In plugins/modules/elastic_pkgpolicy.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765103321>:

> +        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_name=dict(type='str'),

+        agent_policy_id=dict(type='str'),

+        integration_name=dict(type='str', Required=True),

+        pkg_policy_name=dict(type='str', Required=True),

Required should be required

________________________________

In plugins/modules/elastic_pkgpolicy_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765107668>:

> +import json

+

+try:

+  from ansible_collections.expedient.elastic.plugins.module_utils.ece import ECE

+except:

+  import sys

+  import os

+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'

+  sys.path.append(util_path)

+  from ece import ECE

+

ECE is imported but never used

________________________________

In plugins/modules/elastic_pkgpolicy_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765108172>:

> +# Licensed under the Apache License, Version 2.0 (the "License");

+# you may not use this file except in compliance with the License.

+# You may obtain a copy of the License at

+#

+#     http://www.apache.org/licenses/LICENSE-2.0

+#

+# Unless required by applicable law or agreed to in writing, software

+# distributed under the License is distributed on an "AS IS" BASIS,

+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.

+# See the License for the specific language governing permissions and

+# limitations under the License.

+

+from ansible.module_utils.basic import _ANSIBLE_ARGS, AnsibleModule

+#from ansible.module_utils.basic import *

+

+import json

json is imported but never used

________________________________

In plugins/modules/elastic_pkgpolicy.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765108818>:

> +try:

+  from ansible_collections.expedient.elastic.plugins.module_utils.ece import ECE

+except:

+  import sys

+  import os

+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'

+  sys.path.append(util_path)

+  from ece import ECE

ECE is imported but never used

________________________________

In plugins/modules/elastic_endpoint_security.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765110647>:

> +        prebuilt_rules_activate=dict(type='bool', default=True),

+        state=dict(type='str', default='present')

+    )

+    argument_dependencies = []

+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),

+        #('alert-type', 'metrics_threshold', ('conditions'))

+

+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)

+    state = module.params.get('state')

+

+    if module.check_mode:

+        results['changed'] = False

+    else:

+        results['changed'] = True

+

+    kibana = Kibana(module)

In order to get the params defined in SecurityBaseline init, Kibana should be replaced here with SecurityBaseline

________________________________

In plugins/modules/elastic_endpoint_security.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765112171>:

> +          pkg_policy_object_json = json.dumps(pkg_policy_object)

+          pkg_policy_update = kibana.update_pkg_policy(pkg_policy_object_id,pkg_policy_object_json)

The send_api_request takes care of json.dumps so this isn't needed.

________________________________

In plugins/modules/elastic_endpoint_security.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765112905>:

> +try:

+  from ansible_collections.expedient.elastic.plugins.module_utils.ece import ECE

+except:

+  import sys

+  import os

+  util_path = new_path = f'{os.getcwd()}/plugins/module_utils'

+  sys.path.append(util_path)

+  from ece import ECE

ECE is imported but never used

________________________________

In plugins/modules/elastic_pkgpolicy.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765119696>:

> +        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_name=dict(type='str'),

+        agent_policy_id=dict(type='str'),

+        integration_name=dict(type='str', Required=True),

+        pkg_policy_name=dict(type='str', Required=True),

+        pkg_policy_desc=dict(type='str'),

+        state=dict(type='str', default='present')

+    )

+    argument_dependencies = []

+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),

+        #('alert-type', 'metrics_threshold', ('conditions'))

+

+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)

Since a name or id is required as an input, the following logic would ensure the requirement is met.

module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,

                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],

                           required_one_of=[('agent_policy_name', 'agent_policy_id')])

________________________________

In plugins/modules/elastic_endpoint_security.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765121613>:

> +        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_id=dict(type='str'),

+        agent_policy_name=dict(type='str'),

+        integration_name=dict(type='str', Required=True),

+        pkg_policy_name=dict(type='str', Required=True),

+        pkg_policy_desc=dict(type='str'),

+        endpoint_security_antivirus=dict(type='bool', default=True),

+        prebuilt_rules_activate=dict(type='bool', default=True),

+        state=dict(type='str', default='present')

+    )

+    argument_dependencies = []

+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),

+        #('alert-type', 'metrics_threshold', ('conditions'))

+

+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)

Since a name or id is required as an input, the following logic would ensure the requirement is met.

module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,

                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],

                           required_one_of=[('agent_policy_name', 'agent_policy_id')])

________________________________

In plugins/modules/elastic_agentpolicy_info.py<https://github.com/Expedient/ansible-collection-elastic/pull/8#discussion_r765123108>:

> +

+    module_args=dict(

+        host=dict(type='str',Required=True),

+        port=dict(type='int', default=9243),

+        username=dict(type='str', Required=True),

+        password=dict(type='str', no_log=True, Required=True),

+        verify_ssl_cert=dict(type='bool', default=True),

+        agent_policy_name=dict(type='str'),

+        agent_policy_id=dict(type='str')

+    )

+

+    argument_dependencies = []

+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),

+        #('alert-type', 'metrics_threshold', ('conditions'))

+

+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)

Since a name or id is required as an input, the following logic would ensure the requirement is met.

module = AnsibleModule(argument_spec=module_args, supports_check_mode=True,

                           mutually_exclusive=[('agent_policy_name', 'agent_policy_id')],

                           required_one_of=[('agent_policy_name', 'agent_policy_id')])

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Expedient/ansible-collection-elastic/pull/8#pullrequestreview-826745696>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWA6XJPP2N6K4HZ6MEGCAZTUP6P7HANCNFSM5JUG3LJA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

exp-hc commented on plugins/module_utils/kibana.py:
```diff
@@ -28,6 +28,12 @@ def __init__(self, module):
     self.validate_certs = module.params.get('verify_ssl_cert')
     self.version = None # this is a hack to make it so that we can run the first request to get the clutser version without erroring out
     self.version = self.get_cluster_version()
+    self.integration_name = self.module.params.get('integration_name')
+    self.agent_policy_id = self.module.params.get('agent_policy_id')
+    self.agent_policy_name = self.module.params.get('agent_policy_name')
+    self.agent_policy_desc = self.module.params.get('agent_policy_desc')
+    self.pkg_policy_name = self.module.params.get('pkg_policy_name')
+    self.pkg_policy_desc = self.module.params.get('pkg_policy_desc')
```
Sorry, I realize my last comment on this wasn't clear. These should go in the `init` inside the module class **unless** they are used by all modules that use this module_util.

ianrscottexp replied:
I am not sure I understand what you are asking for.  Please check code and provide feedback.

exp-hc commented on plugins/modules/elastic_agentpolicy.py:
```diff
@@ -29,12 +29,12 @@
-        host=dict(type='str',Required=True),
+        host=dict(type='str',required=True),
         port=dict(type='int', default=9243),
         username=dict(type='str', Required=True),
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -90,15 +79,15 @@ def create_securityctrl_baseline_settings(self,pkg_policy_object):
-        host=dict(type='str',Required=True),
+        host=dict(type='str',required=True),
         port=dict(type='int', default=9243),
         username=dict(type='str', Required=True),
```
`Required` should be `required`

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -0,0 +1,164 @@
+        
+    def create_securityctrl_baseline_settings(self,pkg_policy_object):
+        ################ Checking and creating package policy associated with Integration
+        kibana = Kibana(self.module)
```
`Kibana` already exists here as `self` so all instances of `kibana` can be replaced with `self` inside this function.

exp-hc commented on plugins/modules/elastic_endpoint_security.py:
```diff
@@ -0,0 +1,164 @@
+          results['pkg_policy_status'] = "No Integration Package found, Package Policy not created becans check_mode is set to true"
+          results['changed'] = False
+              
+    ElasticSecurityBaseline = SecurityBaseline(module)
```
`SecurityBaseline` is already defined as `kibana` on line 111.

exp-hc reviewed (changes requested):
A few more concerns identified in the comments. Also, have these modules been tested since any of the recent changes?

exp-hc reviewed (approved):
LGTM. Thanks for all the updates
